nutils/evaluable.py

```diff
@@ -134,6 +134,10 @@ class ExpensiveEvaluationWarning(warnings.NutilsInefficiencyWarning):
     pass
 
 
+def _abstract_property(self):
+    raise NotImplementedError(f'{self.__class__.__name__} fails to implement dependencies')
```

Comment:
The error message refers to some property `dependencies`, but `_abstract_property` is used from `shape` and `dtype`.

Reply:
I added an `abstract_property` descriptor class in `_util` that gets hold of the attribute name via `__set_name__`.